File: Array/two-sum.py
# Given an array of integers, return the indices of two numbers that add up to a given target.
import numpy as np

arr = np.array([1, 6])

target = 7 

for p1 in range(len(arr)):
    for p2 in range(p1 + 1, len(arr)):
        if (arr[p1] + arr[p2]) == target:
            print("FOUND: p1 - ", p1, " p2 - ", p2)

File: Array/two-sum.java
// Given an array of integers, return the indices of two numbers that add up to a given target.

class Solution {
    public int[] twoSum(int[] nums, int target) {
        for (int p1 = 0; p1 < nums.length; p1++) {
            int numToFind = target - nums[p1];

            for (int p2 = p1+1; p2 < nums.length; p2++) {
                if (numToFind == nums[p2]) {
                    return new int[] {p1, p2}; 
                }
            }
        }
        return null;
    }
}

// Second Way

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numsMap = new HashMap<>();

        for (int p = 0; p < nums.length; p++) {
            Integer currentMapVal = numsMap.get(nums[p]);

            if (currentMapVal != null && currentMapVal >= 0) {
                return new int[] {currentMapVal, p};
            }

            else {
                int numsToFind = target - nums[p];
                numsMap.put(numsToFind, p);
            }
        }
        return null;
    }
}